### PR TITLE
Update documentation style & fix room version heading order

### PR DESCRIPTION
**Pull request by turt2live**
Also add a missing signing key section to v6.

This additionally contains various edits to the fragments to have them make a little bit more sense in context.

Finally, this updates various aspects of the documentation style which haven't previously been considered - they're added here considering we're in the area.

Fixes https://github.com/matrix-org/matrix-doc/issues/3564
Fixes https://github.com/matrix-org/matrix-doc/issues/3656

Preview: https://pr3683--matrix-org-previews.netlify.app

### CODE REVIEW

File: meta/documentation_style.rst [hunk 1]
```diff
@@ -101,3 +119,61 @@ When writing OpenAPI specifications for the API endpoints, follow these rules:
...
+* A brief (one or two line) description of what the room version is expected to
+  consist of. For example, "..., adding new redaction rules to cover memberships".
+
+The next section must then be "Client considerations" to help client developers avoid
+the naturally complex "Server implementation components" later on. This section must:
+
+* Clearly describe any and all changes which affect users of the Client-Server API.
+* Clearly make reference to the redaction rules. A copy/paste example of this is in
+  Room Version 2: "Clients which implement the redaction algorithm locally should
+  refer to the [redactions] section below for a full overview of the algorithm."
```
Comment (richvdh) [inline meta/documentation_style.rst:144]: I'm not entirely convinced it does need calling out for every single room version, but if we're going to do so, maybe another room version which doesn't change it (such as 3) is a better example:

> Though unchanged in this room version, clients which implement the redaction algorithm locally should refer to the [redactions] section below for a full overview.

**Review by richvdh — APPROVED** [review]
lgtm in general

File: meta/documentation_style.rst (hunk 1 above)
Comment (turt2live) [reply meta/documentation_style.rst:144]: I'm also not convinced it needs to be mentioned, but the link helped client folk feel like they could bypass the server components so might as well include it.

File: meta/documentation_style.rst [hunk 2]
```diff
@@ -19,8 +19,7 @@ nested titles (h6, or 6 `#` characters) and instead re-evaluate the document str
 Correct capitalisation for long section names
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Headings should start with a capital letter, and use lower-case otherwise. This
-document is an example of what we mean.
+Headings should be in sentence case, as represented by this document.
```
[the remark this answers is not in the document]
Comment (turt2live) [reply meta/documentation_style.rst:22]: note for self: linking to https://apastyle.apa.org/style-grammar-guidelines/capitalization/sentence-case is probably fine

**Pull request merged**